core/src/refx/subs.cljc

```diff
@@ -5,6 +5,28 @@
 
 (def kind :sub)
 
+;; --- batch processing -------------------------------------------------------
+
+(defonce ^:private batch-notify-counter (atom 0))
+(defonce ^:private batch-notify-fns (atom []))
+
+(defn start-batch-update! []
+  (swap! batch-notify-counter inc))
+
+(defn end-batch-update! []
+  (swap! batch-notify-counter dec)
+  (when (zero? @batch-notify-counter)
+    (let [notify-fns @batch-notify-fns]
+      (reset! batch-notify-fns [])
+      (doseq [notify-fn notify-fns]
+        (notify-fn)))))
+
+(defn- batch-notify [listeners]
+  (letfn [(notify-fn [] (.notify listeners))]
+    (if (pos? @batch-notify-counter)
+      (swap! batch-notify-fns conj notify-fn)
+      (notify-fn))))
```

Comment:
The concurrency code is not idiomatic. It will work for ClojureScript, because there are no threads there, but might fail for Clojure (granted this is not a huge use case right now).

Instead of: 

```clojure
  (swap! batch-notify-counter dec)
  (when (zero? @batch-notify-counter)
```

we should atomically swap and check the return value:

```clojure
  (when (zero? (swap! batch-notify-counter dec))
```

Having two atoms for the counter and the fns is also problematic:

```clojure
    (if (pos? @batch-notify-counter)
      (swap! batch-notify-fns conj notify-fn)
```

With real parallelism, `(pos? @batch-notify-counter)` could be false in this thread, but another thread could `dec` it to zero and reset the fns, so that the `notify-fn` here could be dropped or enqueued for the next batch, whatever wins the race.

Probably better to use a single `(atom {:counter 0, :fns []})`.

I know this is pretty pedantic, especially since the `:mutable` listeners are not designed to work in JVM-Clojure (refx.alpha, after all). But maybe we should do it right anyway :)